### PR TITLE
Support setting zsh version when including class

### PR DESCRIPTION
**Pull request by salimane**
cc @wfarr @fromonesrc

### CODE REVIEW

**Comment by salimane** [comment]
@jacobbednarz please could you check this ?

**Comment by jacobbednarz** [comment]
Personally, I would have split these two changes up as it looks like there is some bumping of dependencies and a change to the puppet module itself. That way, if we needed to revert one or the other, we won't need to revert the whole lot or formulate a manual revert of part of the changes.

File: manifests/init.pp [hunk 1]
```diff
@@ -4,10 +4,10 @@
 # Examples
 #
 #   include zsh
-class zsh {
+class zsh ($ensure = 'present') {
   require boxen::config
 
-  package { 'zsh': }
+  package { 'zsh': ensure => $ensure }
```
Comment (jacobbednarz) [inline manifests/init.pp]: Do we need to be explicit about the provider here? I.e.

```
package { 'zsh': 
  ensure => $ensure,
  provider => 'homebrew'
}
```

**Comment by salimane** [comment]
@jacobbednarz I have split the commits :)

**Comment by jacobbednarz** [comment]
Looks good to me!

**Comment by jacobbednarz** [comment]
Released as [1.0.1](https://github.com/boxen/puppet-zsh/releases/tag/1.0.1). Thanks! :cake:

**Pull request merged**